### ISSUE TITLE
key-chord broken

### ISSUE THREAD

**Issue by maurelio1234** [lastquestion/explain-pause-mode#7]
Hi,

First of all, thanks for your package :+1: 

I am trying to use this package with `key-chord` (https://github.com/emacsorphanage/key-chord).

I noticed that may key-chord bindings do not work anymore when your package is active. I also notice that it only happens when the operation behind the chord takes a long time (maybe more that your default 40ms setting?). 

There's goes a minimal `test.el` file you use to test this. It uses straight.el to install both your package and key chord

```elisp
(defvar bootstrap-version)
(let ((bootstrap-file
       (expand-file-name "straight/repos/straight.el/bootstrap.el" user-emacs-directory))
      (bootstrap-version 5))
  (unless (file-exists-p bootstrap-file)
    (with-current-buffer
        (url-retrieve-synchronously
         "https://raw.githubusercontent.com/raxod502/straight.el/develop/install.el"
         'silent 'inhibit-cookies)
      (goto-char (point-max))
      (eval-print-last-sexp)))
  (load bootstrap-file nil 'nomessage))

(straight-use-package 'key-chord)

(defun slow-operation ()
  (interactive)
  (sleep-for 2)
  (message "Done!"))

(require 'key-chord)
(key-chord-mode 1)
(key-chord-define-global "ab" 'slow-operation)


(when load-explain
  (message "Loading explain...")
  (straight-use-package '(explain-pause-mode
			  :host github
			  :repo "lastquestion/explain-pause-mode"
			  :branch "master"))
  (require 'explain-pause-mode)
  (explain-pause-mode 1))
```

You can test two scenarios:

* `emacs -Q --eval "(setq load-explain nil)" --load test.el`: baseline: key chord without explain: you can type both keys `ab` and it will run the `slow-operation` and display `Done`
* `emacs -Q --eval "(setq load-explain tl)" --load test.el`: bug: key chord with explain: you can type both keys `ab` and it will never run the `slow-operation`.

I tried increasing `explain-pause-blocking-too-long-ms` but it doesn't seem to help.

**Comment by lastquestion** [comment]
Ahhaha, thank you for the bug report, it's very detailed, and you've made a repro case. 💯 

I have an idea where it is, 99% it's because it is trying to record the long execution in the hash map of candidate profiles somewhere it is breaking, maybe keymaps have a different command call stack. That makes sense.

**Comment by lastquestion** [comment]
Hey @maurelio1234 , I tested the repro case. What I see in `*Messages*` is this:

```
For information about GNU Emacs and the GNU system, type C-h C-a.
Key Chord mode on
Loading explain...
Done!
Emacs blocked for 2000 ms - check *explain-pause-log*
```

The thing is, your message is being overwritten by the long pause message immediately, so you can't see the success of the test command. 

However, I just now fixed a pretty important bug in #11. So if you might try your actual keychord commands again and let me know how that goes, that would be useful. If many of your commands print output to the minibuffer or echo area when they finish, please try setting `explain-pause-alert-via-message` via customize and that will stop printing stuff there.

A message will still get printed when profiling happens, though - #13 will  address that.

**Comment by maurelio1234** [comment]
Hi @lastquestion !

Thank you very much for your prompt response!
Yes I confirm it is working now after your fix :)

BTW, the point of my test was trying to reproduce the bug I saw, in real life I have more mundane things behind my chords like: `ivy-switch-buffer`, `save-buffer`, `magit-status`...

Things that may take time, but that do more than writing messages ;)
